server/v2/store/commands.go

```diff
@@ -0,0 +1,144 @@
+- everything: 2 latest states will be kept
+- custom: allow pruning options to be manually specified through 'pruning-keep-recent'
+
+Note: When the --app-db-backend flag is not specified, the default backend type is 'goleveldb'.
+Supported app-db-backend types include 'goleveldb', 'rocksdb', 'pebbledb'.`,
+		Example: fmt.Sprintf("%s prune custom --pruning-keep-recent 100 --app-db-backend 'goleveldb'", "<appd>"),
+		Args:    cobra.RangeArgs(0, 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// bind flags to the Context's Viper so we can get pruning options.
+			vp := serverv2.GetViperFromCmd(cmd)
+			if err := vp.BindPFlags(cmd.Flags()); err != nil {
+				return err
+			}
+			if err := vp.BindPFlags(cmd.PersistentFlags()); err != nil {
+				return err
+			}
+
+			logger := log.NewLogger(cmd.OutOrStdout())
+			home, err := cmd.Flags().GetString(serverv2.FlagHome) // should be FlagHome
+			if err != nil {
+				return err
+			}
+
+			rootStore, err, interval := createRootStore(cmd, home, vp, logger)
+			if err != nil {
+				return fmt.Errorf("can not create root store %w", err)
+			}
+
+			latestHeight, err := rootStore.GetLatestVersion()
+			if err != nil {
+				return err
+			}
+
+			// valid heights should be greater than 0.
+			if latestHeight <= 0 {
+				return fmt.Errorf("the database has no valid heights to prune, the latest height: %v", latestHeight)
+			}
+
+			upTo := latestHeight - interval
+			cmd.Printf("pruning heights up to %v\n", upTo)
+
+			err = rootStore.Prune(latestHeight)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("successfully pruned the application root multi stores")
+			return nil
+		},
+	}
+
+	cmd.Flags().String(FlagAppDBBackend, "", "The type of database for application and snapshots databases")
+	cmd.Flags().Uint64(FlagPruningKeepRecent, 0, "Number of recent heights to keep on disk (ignored if pruning is not 'custom')")
+	
+	return cmd
+}
+
+func createRootStore(cmd *cobra.Command, rootDir string, v *viper.Viper, logger log.Logger) (storev2.RootStore, error, uint64) {
+	tempViper := v
+	scRawDb, err := db.NewGoLevelDB("application", filepath.Join(rootDir, "data"), nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// handle KeepRecent & Interval flags
+	if cmd.Flags().Changed(FlagPruningKeepRecent) {
+		keepRecent, err := cmd.Flags().GetUint64(FlagPruningKeepRecent)
+		if err != nil {
+			return nil, err, 0
+		}
+
+		// viper has an issue that we could not override subitem then Unmarshal key
+		// so we can not do viper.Set() as comment below
+		// https://github.com/spf13/viper/issues/1106
+		// Do it by a hacky: overwrite app.toml file then read config again.
+
+		// v.Set("store.options.sc-pruning-option.keep-recent", keepRecent) // entry that read from app.toml
+		// v.Set("store.options.ss-pruning-option.keep-recent", keepRecent)
+
+		err = overrideKeepRecent(filepath.Join(rootDir, "config"), keepRecent)
```

Comment:
Have to do a hacky way cause https://github.com/spf13/viper/issues/1106.
If cmd has `keep-recent` flag, overwrite `app.toml` then read it again